samples/uploadWithoutAuth.js

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function main(
+  bucketName = 'my-bucket',
+  contents = 'these are my file contents',
+  destFileName = 'file.txt'
+) {
+  // [START storage_upload_without_authentication]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The contents that you want to upload
+  // const contents = 'these are my contents';
+
+  // The new ID for your GCS file
+  // const destFileName = 'your-new-file-name';
+
+  // Imports the Google Cloud Node.js client library
+  const {Storage} = require('@google-cloud/storage');
+
+  const fetch = require('node-fetch');
+
+  // Creates a client
+  const storage = new Storage();
+
+  async function uploadWithoutAuthentication() {
+    const file = storage.bucket(bucketName).file(destFileName);
+    let location; // endpoint to which we should upload the file
+
```

Comment:
I'm wondering if this sample should be split into two separate samples as to not overload, but it's readable so not completely against it.

Reply:
I think if we separate this into two samples, we run the risk of someone missing one of the samples and only learning one way it could be done. It also depends on how the page is structured though. @cbonnie do you have thoughts here?

Reply:
We can definitely split this up into two samples and make it clear (in writing) that there's two ways to do the task.

Reply:
You could nest them in the same region block: https://cloud.google.com/storage/docs/listing-objects#code-samples is an example

Reply:
oh cool! i didnt realize that. thanks for this example. i will make this change

Reply:
@frankyn @cbonnie please take another look